mlir/lib/Transforms/lab4/kulagin_aleksandr/KulaginAleksandrFMAPass.cpp

```diff
@@ -0,0 +1,91 @@
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassRegistry.h"
+#include "mlir/Support/TypeID.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+
+namespace {
+class KulaginAleksandrFMAPass
+    : public mlir::PassWrapper<KulaginAleksandrFMAPass,
+                               mlir::OperationPass<mlir::ModuleOp>> {
+public:
+  static const char *pluginName;
+  static const char *pluginVersion;
+  mlir::StringRef getArgument() const final { return pluginName; }
+  mlir::StringRef getDescription() const final {
+    return "A pass that merges multiplication and addition into a single "
+           "math.fma";
+  }
+  void runOnOperation() override {
+    mlir::ModuleOp module = getOperation();
+    module.walk([this](mlir::Operation *op) {
+      if (mlir::LLVM::FAddOp addOp = mlir::dyn_cast<mlir::LLVM::FAddOp>(op)) {
+        mlir::Value lhv = addOp.getOperand(0);
+        mlir::Value rhv = addOp.getOperand(1);
+        if (mlir::LLVM::FMulOp defOpLH =
```

Comment:
```suggestion
        if (auto defOpLH =
```

Reply:
In other places please also use `auto` to reduce type specification duplication.